build-utils/interpolate-all-results.js

```diff
@@ -0,0 +1,49 @@
+var parseEnvsVersions = require("./parse-envs-versions");
+
+module.exports = function interoplateAllResults(tests, envs) {
+  var envsTree = buildEnvsTree(envs);
+
+  tests.forEach(function (t) {
+    // Calculate the result totals for tests which consist solely of subtests.
+    if ("subtests" in t) {
+      t.subtests.forEach(function(e) {
+        interpolateTestResults(e.res, envsTree);
+      });
+    } else {
+      interpolateTestResults(t.res, envsTree);
+    }
+  });
```

Comment:
this might be a bit simpler, and won't repeat the interpolate call?
```suggestion
    [].concat(t.subtests || t.res).forEach(function(e) {
      interpolateTestResults(e.res, envsTree);
    });
  });
```